File: 백준/Bronze/10163. 색종이/색종이.py
N = int(input())
board = [[0] * 1001 for _ in range(1001)]

for num in range(1, N+1):
    c, r, w, h = map(int, input().split())
    for i in range(h):
        for j in range(w):
            board[1000-r-i][c+j] = num

answer = [0] * (N+1)
for i in range(1001):
    for j in range(1001):
        for k in range(1, N+1):
            if board[i][j] == k:
                answer[k] += 1
                break

for i in range(1, N+1):
    print(answer[i])

File: 백준/Bronze/10163. 색종이/색종이.java
import java.io.*;
import java.util.StringTokenizer;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null;
        int N = Integer.parseInt(br.readLine());
        int x = 0;
        int y = 0;
        int w = 0;
        int h = 0;
        int[][] board = new int[1001][1001];

        int[] answer = new int[N+1];

        for (int n = 1; n <= N; n++) {
            st = new StringTokenizer(br.readLine());
            x = Integer.parseInt(st.nextToken());
            y = Integer.parseInt(st.nextToken());
            w = Integer.parseInt(st.nextToken());
            h = Integer.parseInt(st.nextToken());

            for (int i = x; i < x + w; i++) {
                for (int j = y; j < y + h; j++) {
                    board[i][j] = n;
                }
            }
        }

        for (int i = 0; i < 1001; i++) {
            for (int j = 0; j < 1001; j++) {
                answer[board[i][j]]++;
            }
        }

        StringBuilder sb = new StringBuilder();

        for (int i = 1; i <= N; i++) {
            sb.append(answer[i]).append("\n");
        }
        System.out.println(sb);
    }
}
